### PR TITLE
Adds the Ethereal dance grenade (properly this time)

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Adds in a new item for Ethereal traitors, the Ethereal Dance Grenade. When thrown the device will create an ethereal disco ball and spew out some micro-disco grenades. These micro disco grenades will cause nearby players to be immobilised, become high and either: Spin, Flip, Dance, Trip or say a random message along the lines of "This party is great!". This item has no effect on Ethereals or mindshielded people and anyone nearby with social anxiety will drop their items to cover their ears. This item probably won't have much use in-game due to its high cost and limited function, but is quite useful for area denial and is a bit of silly fun.
This adds 2 new states to the grenade.dmi file, disco and disco_active which are the textures for the item.
uplink_items.dm is edited to have the item included in traitor uplinks, for Ethereals only.
discogrenade.dm is a brand new file with the code for the device's function.

Additional credits:
 - Creators of Flashbang file for some of the basic grenade function
 - Creators of Ethereal disco ball for light code

## Why It's Good For The Game

New race-specific item for Ethereals traitors, meaning they actually have something in the uplink menu now!

## Changelog
:cl:
tweak: Made the sub-grenade explosion quieter
tweak: Only 1 disco ball is created, rather than >20 which filled the hallways
tweak: The chat message is now sent less frequently to prevent a ridiculous level of chat spam.
balance: Mindshielded persons are now unaffected
tweak: Non-Ethereals with the social anxiety trait will now drop their items and say a message showing their discomfort.
imageadd: Added disco and disco_grenade to grenade.dmi
add: Adds dance grenade to dmi so code can compile
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Could require reduction in cost, but don't want to make it too powerful before it is used in actual missions rather than just thrown at groups of felinids

File: code/game/objects/items/grenades/discogrenade.dm [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	update_mob()
+	var/current_turf = get_turf(src)
+	if(!current_turf)
+		return
+
+	playsound(current_turf, 'sound/weapons/flashbang.ogg', 100, TRUE, 8, 0.9)
+
+	new /obj/structure/etherealball(current_turf)
+
+	for(var/i = 0; i < 6; i ++)
```
Comment (qwertyquerty) [inline code/game/objects/items/grenades/discogrenade.dm]: Change to `for(var/i in 1 to 6)`

**Comment by qwertyquerty** [comment]
This looks good to me. Tested?

**Comment by PowerfulBacon** [comment]
> 
> 
> This looks good to me. Tested?

Tested, doesn't affect mindshielded or Ethereals as intended, I kind of want to change some of the text to make it less spammy, since getting caught in the range will create a lot of messages

File: code/game/objects/items/grenades/discogrenade.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/discogrenade.dm]: Completed

File: code/game/objects/items/grenades/discogrenade.dm [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+//////////////////////
+
+/obj/item/grenade/discogrenade
+	name = "Ethereal Disco Grenade"
+	desc = "An unethical micro-party that will make all non-Ethereal beings dance to its beat!"
+	icon_state = "disco"
+	item_state = "disco"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+	var/list/messages = list("This party is great", "Wooo!!!", "Party!", "Check out these moves!", "Hey, want to dance with me?")
```
Comment (ATH1909) [inline code/game/objects/items/grenades/discogrenade.dm]: add a period or exclamation mark to the end of "This party is great"

File: code/game/objects/items/grenades/discogrenade.dm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	var/list/message_social_anxiety = list("I want to go home...", "Where are the toilets?", "I don't like this song")
```
Comment (ATH1909) [inline code/game/objects/items/grenades/discogrenade.dm]: add a period or an ellipsis to the end of "I don't like this song"

File: code/game/objects/items/grenades/discogrenade.dm [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		return
+
+	if(M.has_quirk(/datum/quirk/social_anxiety))
+		M.say(pick(message_social_anxiety))
+		if(rand(3) && M.get_ear_protection() == 0)
+			M.drop_all_held_items()
+			M.show_message("<span class='warning'>You cover your ears, the music is just too loud for you.</span>", 2)
+		return
+
+	if(HAS_TRAIT(M, TRAIT_MINDSHIELD))
```
Comment (ATH1909) [inline code/game/objects/items/grenades/discogrenade.dm:107]: is this check intended to be after the social anxiety one?

File: code/game/objects/items/grenades/discogrenade.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/discogrenade.dm]: Done

File: code/game/objects/items/grenades/discogrenade.dm (hunk 3 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/discogrenade.dm]: Done

File: code/game/objects/items/grenades/discogrenade.dm (hunk 4 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/discogrenade.dm:107]: Yes, my logic here is that the music would be heard whether you have a mindshield or not, so the social anxiety would come into affect either way.

File: code/game/objects/items/grenades/discogrenade.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/game/objects/items/grenades/discogrenade.dm]: Done

**Comment by ghost** [comment]
lol this is epic im porting this to tg

**Comment by PowerfulBacon** [comment]
> lol this is epic im porting this to tg

Ayy, make sure to port the TC reduction too cause it's way to high in this PR

File: code/game/objects/items/grenades/discogrenade.dm [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+/obj/item/grenade/discogrenade/subgrenade/proc/forcedance(turf/T , mob/living/carbon/human/M)
+	if(!T)
+		return
+	if(M.stat != CONSCIOUS)	//Only conscious people can dance
+		return
+	if(!M || isethereal(M))	//Non humans and non etherals can't dance
+		return
+
+	var/distance = max(0,get_dist(get_turf(src),T))
```
Comment (PowerfulBacon) [inline code/game/objects/items/grenades/discogrenade.dm:96]: Check returned: Line 96: Formatting warning: Commas should have a space after them. (3, 5, 6) not (3,5,6 or 3 ,4 ,5)

**Pull request merged**